docs: move plugin naming (#3874)
vuejs/vue-cli: docs/dev-guide/plugin-dev.md

└── ui.js         # Vue UI integration (optional)
```

## Naming and discoverability

> For a CLI plugin to be usable by other developers, it must be published on [npm](https://www.npmjs.com/) following the name convention `vue-cli-plugin-<name>` or `@scope/vue-cli-plugin-<name>`. Following the name convention allows your plugin to be:

Reviewer: ```suggestion
For a CLI plugin to be usable in a Vue CLI project, it must follow the name convention `vue-cli-plugin-<name>` or `@scope/vue-cli-plugin-<name>`. It allows your plugin to be:
```

- Discoverable by `@vue/cli-service`;
- Discoverable by other developers via searching;
- Installable via `vue add <name>` or `vue invoke <name>`.

> 

Reviewer: Maybe add a warning block here explaining that not following the naming convention will result in problems?

Author: Done 😸
